### PR TITLE
feat(kubectl): Extract interface for KubectlJobExecutor and use it.

### PR DESCRIPTION
**Pull request by DanielaS12**
This change is made to enhance the kubernetes module to add the possibility to decorate the class. This will be usefull when extending the functionalities of the the KubectlJobExecutor in the community or using plugins.

### CODE REVIEW

File: clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/op/job/KubectlJobExecutorInterface.java [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+import com.netflix.spinnaker.clouddriver.kubernetes.description.KubernetesPodMetric;
+import com.netflix.spinnaker.clouddriver.kubernetes.description.manifest.KubernetesKind;
+import com.netflix.spinnaker.clouddriver.kubernetes.description.manifest.KubernetesManifest;
+import com.netflix.spinnaker.clouddriver.kubernetes.security.KubernetesCredentials;
+import com.netflix.spinnaker.clouddriver.kubernetes.security.KubernetesSelectorList;
+import io.kubernetes.client.openapi.models.V1DeleteOptions;
+import java.util.List;
+import javax.annotation.Nonnull;
+
+public interface KubectlJobExecutorInterface {
```
Comment (ovidiupopa07) [inline clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/op/job/KubectlJobExecutorInterface.java:32]: let's rename it to `KubectlJobExecutor` and the implementation to `DefaultKubectlJobExecutor`

File: clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/op/job/KubectlJobExecutor.java [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright 2017 Google, Inc.
+ * Copyright 2023 Netflix, Inc.
```
Comment (ovidiupopa07) [inline clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/op/job/KubectlJobExecutor.java:2]: Armory?

Comment (DanielaS12) [reply clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/op/job/KubectlJobExecutor.java:2]: sure

**Review by ovidiupopa07 — APPROVED** [review]
Approving since this is only adding a marker interface that will improve the extensibility of the class, allowing downstream consumers to customize some parts of the OSS implementation

**Comment by ovidiupopa07** [comment]
@Mergify back-port release-1.28.x

**Pull request merged**